### PR TITLE
op-reth: Fix reth-primitives signature tests

### PR DESCRIPTION
**Pull request by BrianBland**
Updates signature tests in `reth-primitives` to be compatible with the special-case optimism serialization for zero signatures (`R: 0, S: 0, V: 0`).

Adds optimism-only test cases for this zero signature behavior

Fixes the following tests:
```
reth-primitives transaction::signature::tests::test_encode_and_decode_with_eip155_chain_id
reth-primitives transaction::signature::tests::test_payload_len_with_eip155_chain_id
reth-primitives transaction::signature::tests::test_v
```

### CODE REVIEW

File: crates/primitives/src/transaction/signature.rs [hunk 1]
```diff
@@ -157,7 +157,7 @@ mod tests {
 
     #[test]
     fn test_payload_len_with_eip155_chain_id() {
-        let signature = Signature { r: U256::default(), s: U256::default(), odd_y_parity: false };
+        let signature = Signature { r: U256::from(1), s: U256::from(1), odd_y_parity: false };
```
Comment (merklefruit) [inline crates/primitives/src/transaction/signature.rs:160]: could you explain briefly why the change from 0 to 1 here?

Comment (BrianBland) [reply crates/primitives/src/transaction/signature.rs:160]: This is specifically to account for the new `Signature.v()` logic which returns `0` if `r` and `s` are both `0`
https://github.com/anton-rs/op-reth/blob/3cc5b222204a4a1143c8aea6651f3f361b665e5f/crates/primitives/src/transaction/signature.rs#L65-L67

Comment (BrianBland) [reply crates/primitives/src/transaction/signature.rs:160]: Happy to add this as an inline comment if this is deemed appropriate

Comment (rkrasiuk) [reply crates/primitives/src/transaction/signature.rs:160]: please, do. we'll be incredibly meticulous about the changes outside of the feature flag when reviewing the final PR

Comment (BrianBland) [reply crates/primitives/src/transaction/signature.rs]: Updated 👍
